pkg/tsdb/calculator.go

```diff
@@ -0,0 +1,227 @@
+package tsdb
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
```

Comment:
If we're considering moving this code to the SDK we don't want to reference/use the simplejson package, [ref](https://github.com/grafana/grafana/blob/main/pkg/README.md#avoid-use-of-the-simplejson-package).

Shouldn't this use backend.TimeRange rather than plugins.DataTimeRange to be compatible with the SDK?

And tests as you mentioned would be nice as well.